crates/matrix-sdk/src/test_utils/mocks.rs

```diff
@@ -1387,7 +1387,7 @@ impl<'a> MockEndpoint<'a, RoomSendEndpoint> {
     /// assert_eq!("$some_id", response.delay_id);
     /// # anyhow::Ok(()) });
     /// ```
-    pub fn with_delay(self, delay: Duration) -> Self {
+    pub fn match_delay(self, delay: Duration) -> Self {
```

Comment:
Perhaps `match_delayted_event()` would be an even better name, also the docs should really link to the MSC.